evalml/automl/utils.py

```diff
@@ -85,3 +85,30 @@ def tune_binary_threshold(pipeline, objective, problem_type, X_threshold_tuning,
             y_predict_proba = pipeline.predict_proba(X_threshold_tuning)
             y_predict_proba = y_predict_proba.iloc[:, 1]
             pipeline.threshold = objective.optimize_threshold(y_predict_proba, y_threshold_tuning, X=X_threshold_tuning)
+
+
+def check_all_pipeline_names_unique(pipelines):
+    """Checks whether all the pipeline names are unique.
+
+    Arguments:
+        pipelines (list(PipelineBase)): List of pipelines to check if all names are unique.
+
+    Returns:
+          None
+
+    Raises:
+        ValueError if any pipeline names are duplicated.
+    """
+    seen_names = set()
+    duplicate_names = set()
+
+    for pipeline in pipelines:
+        if pipeline.name in seen_names:
+            duplicate_names.add(pipeline.name)
+        else:
+            seen_names.add(pipeline.name)
```

Comment:
I have twice proposed a pandas impl and twice deleted it thinking I was being overbearing.  Here it is...not blocking, feel free to reject.

```
name_count = pd.Series([p.name for p in pipelines]).value_counts()
duplicate_names = name_count[name_count > 1] # I don't think this line is quite right.
```

Reply:
Ended up taking it! 👏